### ISSUE TITLE
Monerod Not Honoring Upload Bandwidth Limit?

### ISSUE THREAD

**Issue by downystreet** [monero-project/monero#7388]
Daemon Version: 0.17.1.9
OS: CentOS 8

When running monerod for longer than several hours I noticed that the bandwidth being used by the daemon is exceeding the default upload limit of 2048 kb/s. The Download rate has been within the limit. I have used several programs to monitor the connection and each one gives somewhat different readings. Monerod is the only program running on the server. When monitoring the server with nload I am seeing upwards of 25 Mbit/s sometimes. I also checked the data from the router and it indicated a reading of approximately 17 Mbit/s from the server. Nethogs gives a much lower upload speed when monitoring but it did go to over 3 Mbit/s which is over the limit of 2048 Mbit/s set in the daemon. I have also been noticing a lot of blocked host and unblocked host in the daemon output. Inbound connections are around 100. Sometimes the out connection will go over the default of 12 and be 13 in the daemon.

**Comment by hyc** [comment]
Is your node doing P2P traffic only, or does it also have RPC enabled? I've noticed that only the P2P traffic honors the bandwidth limits, there is no limiter for RPC sessions.

**Comment by downystreet** [comment]
Yes, RPC is enabled. Nethogs is in Mb/s and the router and nload are Mbit/s. So I guess it is within the limits but I still haven't got an answer for why my miners are getting a connection timeout error from the daemon at random times.

**Comment by cirocosta** [comment]
just for completeness: yes, indeed, throttling is _not_ activated for RPC:

https://github.com/monero-project/monero/blob/a39b1d56c8835798e2f5e3cc43c33b5f2d8e13da/contrib/epee/include/net/abstract_tcp_server2.inl#L930-L933

perhaps that should be more well advertised in the flags' description?

```
  --limit-rate-up arg (=2048)           set limit-rate-up [kB/s]
  --limit-rate-down arg (=8192)         set limit-rate-down [kB/s]
  --limit-rate arg (=-1)                set limit-rate [kB/s]
```

**Comment by hyc** [comment]
I kinda think it ought to be. That's the reason I turned off public RPC on my public node, it was eating my monthly bandwidth quota.

**Comment by asheroto** [comment]
Agreed.... +1 vote for this

**Comment by Jayd603** [comment]
So... is that kB/s as in kBytes/s or kb/s as in Kbits/s (where as the upper case B usually denotes Bytes not bits)?    I set mine to 80000 , which in kBytes/s (upper case B) is 80 MBytes/s or 800 Mbit/s.
